### DISCUSSION TITLE
enable / disable addons with terraform inputs

### DISCUSSION THREAD

**Discussion by swibrow** [tx-pts-dai/terraform-aws-kubernetes-platform#17] [Polls]
what is preferred: enable nested in the lacework input or dedicated enable_lacework input?
```
variable "enable_lacework" {
  description = "Enable Lacework integration"
  type        = bool
  default     = false
}

variable "lacework" {
  description = "Map of Lacework configurations"
  type        = any
  default     = {}
}
```

```
variable "lacework" {
  description = "Map of Lacework configurations"
  type        = any
  default     = {
    enable = false
  }
}
```

**Comment by swibrow** [comment]
I'd say dedicated so that if a custom config is passed, a try() wouldnt need to be used in the module

**Reply by Davidoutz** [reply]
If enabling Lacework is almost always tweaked, nesting might keep things cleaner and bundled, also especially if we end up with a lot of inputs for the platform. On the other hand if we keep most often a default config I would go with dedicated 🤷‍♂️ So I am not sure what to answer yet 😄 Probably we could standardise most of the configuration so the map of config would rarely be used🤔 and then go for dedicated

**Reply by swibrow** [reply]
You're on point for concern of too many inputs. For now i've implemented with dedicated since the map is now full of 'defaults' as I'm not sure of a better way to pass inputs with submodules.
